### PR TITLE
Make TestEmbeddingNNDeviceTypeCPU::test_EmbeddingBag_per_sample_weights_and_no_offsets_cpu_int32_float32 compatible with TorchDynamo

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120831

Previously, the test case directly accesses the tensor data via tensor.data which is not supported on FakeTensor. So we manually copy the tensor as a workaround. 
Fixes: https://github.com/pytorch/pytorch/issues/119788

### CODE REVIEW

File: test/nn/test_embedding.py [hunk 1]
```diff
@@ -1002,9 +1003,9 @@ def _test_EmbeddingBag_vs_Embedding(self, N, D, B, L, max_norm=None,
 
         output.backward(grad_output)
         ref_output.backward(grad_output)
-        es_weight_grad = es.weight.grad.data
+        es_weight_grad = copy.deepcopy(es.weight.grad)
```
Comment (janeyx99) [inline test/nn/test_embedding.py:1006]:
```suggestion
        es_weight_grad = es.weight.grad.detach()
```

or use clone()?  deepcopy is likely not the right thing to do :p

Comment (tugsbayasgalan) [reply test/nn/test_embedding.py]: ok done

File: test/nn/test_embedding.py [hunk 2]
```diff
@@ -1002,9 +1002,9 @@ def _test_EmbeddingBag_vs_Embedding(self, N, D, B, L, max_norm=None,
 
         output.backward(grad_output)
         ref_output.backward(grad_output)
-        es_weight_grad = es.weight.grad.data
+        es_weight_grad = torch.clone(es.weight.grad)
```
Comment (janeyx99) [inline test/nn/test_embedding.py:1005]: Did detach not work lol

**Review by janeyx99 — COMMENTED** [review]
Would trust @mikaylagawarecki's thoughts on whether this tests the same stuff

File: test/nn/test_embedding.py (hunk 2 above)
Comment (tugsbayasgalan) [reply test/nn/test_embedding.py:1005]: detach had a same error :(

Comment (mikaylagawarecki) [inline test/nn/test_embedding.py:1005]: Why did we need the `.data` here initially? 

From running locally, I think just `es.weight.grad` might be ok

**Comment by mikaylagawarecki** [comment]
Separately, I think the other 11 tests when expanding the dropdown in the linked issue might still be failing (probably also due to .data calls in this file) 😅 Would be nice to fix them all!

File: test/nn/test_embedding.py (hunk 2 above)
Comment (mikaylagawarecki) [reply test/nn/test_embedding.py:1005]: For more context, `.data` calls into `t.variable_data`, impl here 

https://github.com/pytorch/pytorch/blob/aa36821615a4b5cfa748b0fe07fb2f1050af629b/torch/csrc/autograd/variable.cpp#L438-L446

It seems like we just compare `es_weight_grad` with `e.weight.grad` below, so I don't see why this initial `.data` call here would be necessary 🤔

**Review by janeyx99 — COMMENTED** [review]
would it be better to disable assertRaises from within the dynamo code instead vs here?

**Comment by tugsbayasgalan** [comment]
> would it be better to disable assertRaises from within the dynamo code instead vs here?

Yepp i think so too. I think we should graph break on assertRaises

**Comment by janeyx99** [comment]
ah i suppose this is also hinged on https://github.com/pytorch/pytorch/pull/119599

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
@pytorchbot merge -r

forcing a rebase to see pull get triggered hopefully
